pkg/util/pod/pod.go

```diff
@@ -56,3 +63,36 @@ func gateIndex(p *corev1.Pod, gateName string) int {
 		return g.Name == gateName
 	})
 }
+
+func ReadUIntFromLabel(obj client.Object, labelKey string) (*int, error) {
+	return ReadUIntFromLabelWithMax(obj, labelKey, math.MaxInt)
+}
+
+func ReadUIntFromLabelWithMax(obj client.Object, labelKey string, max int) (*int, error) {
+	value, found := obj.GetLabels()[labelKey]
+	kind := obj.GetObjectKind().GroupVersionKind().Kind
+	if !found {
+		return nil, fmt.Errorf("no label %q for %s %q", labelKey, kind, klog.KObj(obj))
+	}
+	intValue, err := readUIntFromStringWithMax(value, max)
+	if err != nil {
+		return nil, fmt.Errorf("incorrect label value %q for %s %q: %w", value, kind, klog.KObj(obj), err)
+	}
+	return intValue, nil
+}
+
+var (
+	errInvalidUInt = errors.New("invalid unsigned integer")
+)
+
+func readUIntFromStringWithMax(value string, max int) (*int, error) {
+	uintValue, err := strconv.ParseUint(value, 10, 0)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %s", errInvalidUInt, err.Error())
+	}
+	intValue := int(uintValue)
+	if intValue > max {
+		return nil, fmt.Errorf("value should be less than or equal to %d", max)
```

Comment:
I'm wondering if we can also have here `return nil, fmt.Errorf("%w: value should be less or equal: %d", errInvalidValue, max)`. Then, in the testing we could compare errors rather than strings, IIUC.

Reply:
Done